src/watcher/store.ts

```diff
@@ -116,16 +121,17 @@ export class AppointmentStore extends StartStopService implements IAppointmentSt
             // remove the appointment from the id index
             delete this.appointmentsById[appointmentId];
 
-            // remove the appointment from the state locator index
-            const currentAppointment = this.appointmentsByStateLocator[appointmentById.getStateLocator()];
-            // if it has the same id
-            if (currentAppointment.id === appointmentId) {
-                delete this.appointmentsByStateLocator[appointmentById.getStateLocator()];
-            }
-
-            // and remove from remote storage
-            await this.db.del(appointmentId);
-
+            await this.lockManager.withLock(appointmentById.getStateLocator(), async () => {
+                // remove the appointment from the state locator index
+                const currentAppointment = this.appointmentsByStateLocator[appointmentById.getStateLocator()];
+                // if it has the same id
+                if (currentAppointment.id === appointmentId) {
+                    delete this.appointmentsByStateLocator[appointmentById.getStateLocator()];
+                }
+
+                // and remove from remote storage
+                await this.db.del(appointmentId);
```

Comment:
I dont think this needs to be inside the lock

Reply:
But then, if there is an update to the state locator that leads to a db action adding the same appointmentId, we don't have a guarantee that the insertion happens after `db.del`; if not, it would be immediately deleted on the db, but not in memory.
Am I missing something?

Reply:
As per discussion, removed db.del from the critical section, and added a comment.